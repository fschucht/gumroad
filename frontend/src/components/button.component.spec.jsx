import renderer from 'react-test-renderer';
import { Button } from './button.component';

describe('Button', () => {
  it('should render a Button component', () => {
    const renderedComponent = renderer.create(<Button>Hello World!</Button>).toJSON()

    expect(renderedComponent).toMatchSnapshot()
  })

  describe('when type is primary', () => {
    it('should render a primary Button component', () => {
      const renderedComponent = renderer.create(<Button type='primary'>Hello World!</Button>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when type is secondary', () => {
    it('should render a secondary Button component', () => {
      const renderedComponent = renderer.create(<Button type='secondary'>Hello World!</Button>).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })
})