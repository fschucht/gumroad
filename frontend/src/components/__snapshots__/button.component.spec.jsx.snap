// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Button component 1`] = `
<a
  className="button"
>
  Hello World!
</a>
`;

exports[`Button when type is primary should render a primary Button component 1`] = `
<a
  className="button button--primary"
>
  Hello World!
</a>
`;

exports[`Button when type is secondary should render a secondary Button component 1`] = `
<a
  className="button button--secondary"
>
  Hello World!
</a>
`;
